_posts/2021-04-14-accessing-the-relay-store.markdown

```diff
@@ -0,0 +1,260 @@
+  shallow: { type: "Boolean", defaultValue: false }
+) {
+  id
+  artworksConnection(
+    first: $first
+    after: $after
+    artworkIDs: $checklistArtworkIDs
+    exclude: $checklistExcludedArtworkIDs
+    missingPriorityMetadata: $missingPriorityMetadata
+    publishedWithin: $publishedWithin
+    shallow: $shallow
+  ) @connection(key: "ArtworksMissingMetadata_partner_artworksConnection", filters: []) {
+    totalCount
+    edges {
+      node {
+        internalID
+        ...ArtworksMissingMetadataItem_artwork
+      }
+    }
+  }
+}
+```
+
+## The Damn User
+
+The problem we were running into is that when the user presses “snooze” on an item, Redis is updated but the list
+that we’re using in our Relay query becomes stale. When the user reloads the page, the response from our Relay
+query is correct because it's using the fresh Redis list, but we needed to make sure that the list in the UI is
+also correct and does not include that snoozed item.
+
+Without reloading, our initial solution was to use a local state variable in our React component to keep track of
+which items were snoozed. This looked like:
+
+```js
+const [localSnoozedItems, setLocalSnoozedItems] = useState([])
+```
+
+`localSnoozedItems ` and `setLocalSnoozedItems` were passed down to each of the children items so when the “snooze”
+button was pressed on that item, not only was the new key stored in Redis (for future reloads of the page) but the
+`localSnoozedItems` state variable was also updated. The connection returned from our Relay query (which remember,
+is already filtered based on our Redis `excludeIDs` from Redis) passes through an additional check to make sure
+that none of those items were snoozed locally and checked again the `localSnoozedItemsList`.
+
+This worked, but it definitely did not feel great to have two sources of truth for snoozing: The Redis key and the
+local state variable.
+
+## Head to the Relay Store
+
+Cue the [RelayModernStore][relay-documentation-relay-modern-store]! I learned that Relay keeps track of the GraphQL
+data returned by each query in a store on the client. Each record in the store has a unique ID and the store can be
+changed, added to, and deleted from. There are a couple of helpful blog posts (like
+[this][deep-dive-into-the-relay-store] and
+[this][wrangling-the-client-store-with-the-relay-modern-updater-function]) that explain the store and how to
+interact with it.
+
+In most of the Relay documentation, blog posts, and Artsy’s uses cases, the store is accessed through an updater
+function via [mutations][relay-documentation-mutations]. [Updater functions][relay-documentation-updater-functions]
+that take the store as an argument can optionally be added to Relay mutations. Inside that function, you can use
+the store to access and modify the records you need.
+
+Here's an example from our app:
+
+```js
+commitMutation <
+  MyCollectionArtworkModelDeleteArtworkMutation >
```

Comment:
Ditto on this code example -- we can strip it down to only the bare essentials in order to get the point across. Doesn't even need to contain any artsy-specific references since what we're really getting at is "here's `commitMutation`, it has an `updater` function". Everything else is secondary. 

Also, a more general comment (that I also made on @mdole's post today -- px team is killin it) is that in blog posts, unless one is specifically talking about TypeScript, there's no need to include types. It just adds noise unless done deliberately in aid to the code snippet or post.